presto-main/src/test/java/com/facebook/presto/metadata/TestFunctionInvokerProvider.java

```diff
@@ -0,0 +1,62 @@
+package com.facebook.presto.metadata;
+
+import com.facebook.presto.operator.scalar.AbstractTestFunctions;
+import com.facebook.presto.spi.InvocationConvention;
+import com.google.common.collect.ImmutableList;
+import org.testng.annotations.Test;
+
+import static com.facebook.presto.metadata.FunctionInvokerProvider.checkChoice;
+import static com.facebook.presto.operator.scalar.ScalarFunctionImplementation.NullConvention.BLOCK_AND_POSITION;
+import static com.facebook.presto.operator.scalar.ScalarFunctionImplementation.NullConvention.RETURN_NULL_ON_NULL;
+import static com.facebook.presto.operator.scalar.ScalarFunctionImplementation.NullConvention.USE_BOXED_TYPE;
+import static com.facebook.presto.operator.scalar.ScalarFunctionImplementation.NullConvention.USE_NULL_FLAG;
+import static com.facebook.presto.spi.InvocationConvention.InvocationArgumentConvention.BLOCK_POSITION;
+import static com.facebook.presto.spi.InvocationConvention.InvocationArgumentConvention.BOXED_NULLABLE;
+import static com.facebook.presto.spi.InvocationConvention.InvocationArgumentConvention.NEVER_NULL;
+import static com.facebook.presto.spi.InvocationConvention.InvocationArgumentConvention.NULL_FLAG;
+import static org.testng.Assert.assertFalse;
+import static org.testng.Assert.assertTrue;
+
+public class TestFunctionInvokerProvider
+        extends AbstractTestFunctions
+{
+    @Test
+    public void testFunctionInvokerProvider()
+    {
+        assertTrue(checkChoice(ImmutableList.of(USE_BOXED_TYPE, USE_BOXED_TYPE),
+                true,
+                false,
+                new InvocationConvention(ImmutableList.of(BOXED_NULLABLE, BOXED_NULLABLE),
+                        InvocationConvention.InvocationReturnConvention.BOXED_NULLABLE, false)));
```

Comment:
Always put first argument on its own line when you spread the arguments on multiple lines.

```
assertTrue(checkChoice(
        ImmutableList.of(USE_BOXED_TYPE, USE_BOXED_TYPE),
        true,
        false,
        new InvocationConvention(ImmutableList.of(BOXED_NULLABLE, BOXED_NULLABLE), BOXED_RETURN, false)));